include/effects/matrix/PatternWeather.h

```diff
@@ -298,26 +418,42 @@ class PatternWeather : public LEDStripEffect
             vTaskDelay(pdMS_TO_TICKS(WEATHER_CHECK_WIFI_WAIT));
         }
 
+        if (g_ptrSystem->DeviceConfig().GetOpenWeatherAPIKey().isEmpty())
+        {
+            debugW("No API Key Present!");
+            return;
+        }
+
+
         updateCoordinates();
 
-        if (getWeatherData())
+        if (locationValid)

```

Comment:
The location indicator is either a zip code or a city name. If the lookup of those fail then there is not a valid location.

Reply:
I think it's an incomplete implementation of location validation, though. The `Draw()` method doesn't check for "validLocation", so it will just draw the old weather for the previously successfully converted former location, if there ever was one. 

I'd argue that a comprehensive implementation should reset "dataReady" whenever the location settings are changed, and either lat/long conversion or weather retrieval doesn't succeed. Of course, if such an approach is implemented, depending on what it is that actually failed and why, lat/long conversion and/or weather retrieval should be retried at regular intervals.

Reply:
For clarity: the fact such a "catch and retry" is currently not in place is not your fault. The point I'm merely making is that if we add the complication of (additional) validations, we should do so comprehensively, because it doesn't really serve much of a purpose otherwise. 

Phrased differently: if we add more validations we should do it in such a way that we either draw today's and tomorrow's weather for the currently configured location if everything checks out, or we don't if anything's amiss. If something is wrong, we then also retry if that's sensible, and maybe even report what's wrong on the weather effect display instead of the weather we'd otherwise show.

Reply:
I agree our user base at 0, 0 may be very small indeed, but I also think we can't use values that _are_ valid in their context and _should_ as such yield weather data from the weather API as flags for "invalid".

Reply:
One point I tried to make before is that I think we should refrain from adding additional partial validations (like the location one, for which a new `locationValid` variable is even proposed to be introduced), because it will still leave a validation implementation that's partly broken - just in a different way/place. 

Fully _fixing_ the validation - including implementing sensible debugging and maybe even visually reporting problems to the user(!) is an undertaking in itself, and maybe one we should consider separately and then isolate in a completely separate PR.

Reply:
> I'm sure we'll agree there are better ways of representing "unknown location" that 0, 0.

Yes exclamation mark :)

Reply:
The effect does not store the latitude and longitude just the city name or zip code.  Maybe I will re-do all that shortly.